src/agent/onefuzz-task/src/tasks/coverage/dotnet.rs

```diff
@@ -0,0 +1,345 @@
+            .target_exe(&self.config.target_exe)
+            .target_options(&self.config.target_options)
+            .task_id(&self.config.common.task_id);
+
+        //TODO: ./dotnet-coverage collect --output-format cobertura -o "fullcommand.cobertura.xml" "/usr/bin/dotnet bin/Debug/net6.0/testcoverage.dll input2"
+        let dotnet_coverage_path = dotnet_coverage_path()?;
+        let dotnet_path = dotnet_path()?;
+        let id = Uuid::new_v4();
+        let output_file_path = self.intermediate_coverage_files_path()
+            .join(format!("{}.cobertura.xml", id));
+
+        let mut cmd = Command::new(dotnet_coverage_path);
+        cmd.arg("collect")
+            .args(["--output-format", "cobertura"])
+            .args(["-o", &output_file_path.to_string_lossy()])
+            .arg(format!("{} {} {}", dotnet_path.to_string_lossy(), self.config.target_exe.to_string_lossy(), input.to_string_lossy()));
+
+        let target_options = expand.evaluate(&self.config.target_options)?;
+        cmd.args(target_options);
+
+        for (k, v) in &self.config.target_env {
+            cmd.env(k, expand.evaluate_value(v)?);
+        }
+
+        cmd.env_remove("RUST_LOG");
+        cmd.stdin(Stdio::null());
+        cmd.stdout(Stdio::piped());
+        cmd.stderr(Stdio::piped());
+
+        Ok(cmd)
+    }
+
+    fn working_dir(&self) -> &PathBuf {
+        &self
+            .config
+            .coverage
+            .local_path
+    }
+
+    fn intermediate_coverage_files_path(&self) -> PathBuf {
+        self.working_dir()
+            .join("intermediate-coverage-files")
+    }
+
+    async fn command_for_merge(&self) -> Result<Command> {
+        //TODO: dotnet coverage merge output.cobertura.xml output2.cobertura.xml --output-format cobertura -o "final.cobertura.xml"
+        let dotnet_coverage_path = dotnet_coverage_path()?;
+
+        let output_file = self.working_dir()
+            .join(COBERTURA_COVERAGE_FILE);
+        
+        let mut cmd = Command::new(dotnet_coverage_path);
+        cmd.arg("merge")
+            .args(["--output-format", "cobertura"])
+            .args(["-o", &output_file.to_string_lossy()])
+            .arg("-r")
+            .arg("--remove-input-files")
+            .arg("*.cobertura.xml");
+
+        cmd.current_dir(self.working_dir());
+
+        Ok(cmd)
+    }
+
+    fn uses_input(&self) -> bool {
+        let input = PlaceHolder::Input.get_string();
+
+        for entry in &self.config.target_options {
+            if entry.contains(&input) {
+                return true;
+            }
+        }
+        for (k, v) in &self.config.target_env {
+            if k == &input || v.contains(&input) {
+                return true;
+            }
+        }
+
+        false
+    }
+    
+}
+
+fn spawn_with_timeout(cmd: &mut Command, timeout: Duration) -> Result<()> {
+    let mut child = cmd.spawn()?;
+    let _timer = Timer::new(timeout, move || child.kill());
```

Comment:
You can you [tokio::time::timeout](https://docs.rs/tokio/latest/tokio/time/fn.timeout.html#examples) for this.
Also if you use `tokio::process::Command` instead of `std::process::Command` you can wait asynchronously on the child.

Reply:
you’ll also need to set `cmd.kill_on_drop(true)` so that when the timeout expires the process will be killed